app/src/main/java/lt/andro/metasocket/MainActivity.java

```diff
@@ -40,52 +42,42 @@ protected void onCreate(Bundle savedInstanceState) {
         buttonOn = (Button) findViewById(R.id.button_on);
         buttonOff = (Button) findViewById(R.id.button_off);
 
-        buttonOn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                try {
-                    Gpio gpioModule = mwBoard.getModule(Gpio.class);
-                    gpioModule.clearDigitalOut(GPIO_PIN);
-                } catch (UnsupportedModuleException e) {
-                    e.printStackTrace();
-                }
-            }
-        });
+        View.OnClickListener turnOn = new PullSocket(Gpio.PullMode.PULL_UP);
+        View.OnClickListener turnOff = new PullSocket(Gpio.PullMode.PULL_DOWN);
 
-        buttonOff.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                try {
-                    Gpio gpioModule = mwBoard.getModule(Gpio.class);
-                    gpioModule.setDigitalOut(GPIO_PIN);
-                } catch (UnsupportedModuleException e) {
-                    e.printStackTrace();
-                }
-            }
-        });
+        buttonOn.setOnClickListener(turnOn);
+        buttonOff.setOnClickListener(turnOff);
+    }
 
-        ///< Bind the service when the activity is created
-        getApplicationContext().bindService(new Intent(this, MetaWearBleService.class),
-                this, Context.BIND_AUTO_CREATE);
+    @Override
+    protected void onPause() {
+        super.onPause();
+        ///< Unbind the service when the activity is hidden
```

Comment:
See this [comment](https://github.com/andro-lt/metasocket/pull/1#pullrequestreview-9407520). That's a copy of previous comment.